### PR TITLE
Add custom namespace support

### PR DESCRIPTION
**Pull request by anishasthana**
Jira: https://issues.redhat.com/browse/ODH-419

This PR will allow us to deploy notebooks to a custom namespace. There are two main portions to the PR.

The first is just adding an extra environment variable to the base jupyterhub deployment. Once the next jupyterhub-odh image is out we will be able to specify the namespace that jupyterhub should deploy notebooks to.

The second commit on this PR adds the required rbac for jupyterhub to be able to deploy to the notebook destination namespace. To test this PR, you just need to do the following

1. Create opendatahub and odh-notebooks namespace
2. Modify the jupyterhub dc to refer to the following image `quay.io/anishasthana/jupyterhub-img:test-13` (It's built from the latest jupyterhub-odh master)
3.  Modify the value of `notebook_destination` under `jupyterhub/jupyterhub/base/params.env` to be `odh-notebooks`
4. Create a kfdef in opendatahub that is deploying `jupyterhub/jupyterhub` (make sure it is pulling the changes from this PR)
5. Create a kfdef in odh-notebooks that is deploying `jupyterhub/custom-notebook-deployment`

### CODE REVIEW

**Comment by anishasthana** [comment]
@nakfour I've added a description, can we unhold the PR again?

**Comment by nakfour** [comment]
@anishasthana this needs an update to Readme and CI tests

**Comment by anishasthana** [comment]
I've created https://issues.redhat.com/browse/ODH-423 to follow up on this for CI.

**Review by crobby — COMMENTED** [review]
I have gone through the steps to test this and it appears to be working as designed.

**Comment by anishasthana** [comment]
README updates are here as well.

File: jupyterhub/README.md [hunk 1]
```diff
@@ -90,3 +89,38 @@ Contains build chain manifest for CUDA enabled ubi 7 based images, provides `ten
...
 
 *NOTE:* Builds in this overlay require 4-6 GB of memory
+
+## Deploying JupyterHub notebooks to custom namespaces
+
+JupyterHub has support for deploying notebooks to other namespaces. To help facilitate this, we provide a component under `jupyterhub/custom-notebook-deployment`. The steps to deploy are as follows:
+
+1. Create the namespace for JupyterHub to reside in.
+2. Create the namespace for the JupyterHub notebooks to reside in.
+3. Update the value of `notebook_destination` in `jupyterhub/jupyterhub/base/params.env` to be the same as the namespace created in step 2.
```
Comment (crobby) [inline jupyterhub/README.md:99]: I feel like after users do this step (and step 4), they will have difficulty creating a KfDef that points to their changes.  Isn't it possible to specify the parameter values in the KfDef?

Comment (anishasthana) [reply jupyterhub/README.md]: Updated. what do you think?

**Comment by crobby** [comment]
Ok, I have tested with the updated instructions.  Seems good.

**Review by crobby — APPROVED** [review]

**Pull request merged**